workers/unity/Packages/io.improbable.gdk.tools/GenerateCode.cs

```diff
@@ -476,5 +450,94 @@ private static IEnumerable<string> FindDirInPackages(string directory)
 
             return packagePaths.Union(cachedPackagePaths).Distinct();
         }
+
+        internal static void GenerateCodegenRunConfigs(GdkToolsConfiguration toolsConfig = null)
+        {
+            toolsConfig = toolsConfig ?? GdkToolsConfiguration.GetOrCreateInstance();
+
+            var schemaCompilerPath = GetSchemaCompilerPath();
+            var logfilePath = GetDefaultLoggerOutputPath();
+
+            var codegenArgs = new List<string>
+            {
+                $"--json-dir=\"{ImprobableJsonDir}\"",
+                $"--schema-compiler-path=\"{schemaCompilerPath}\"",
+                $"--worker-json-dir=\"{WorkerJsonPath}\"",
+                $"--log-file=\"{logfilePath}\""
+            };
+
+            if (toolsConfig.VerboseLogging)
+            {
+                codegenArgs.Add($"--verbose");
+            }
+
+            var codegenOutputPath = Path.GetFullPath(Path.Combine(Application.dataPath, "..", toolsConfig.CodegenOutputDir));
+            codegenArgs.Add($"--native-output-dir=\"{codegenOutputPath}\"");
+
+            // Add user defined schema directories, warn if directory does not exist
+            foreach (var schemaDir in toolsConfig.SchemaSourceDirs)
+            {
+                if (!Directory.Exists(schemaDir))
+                {
+                    Debug.LogWarning($"Schema directory defined in GDK tools configuration does not exist: {schemaDir}");
+                    continue;
+                }
+
+                codegenArgs.Add($"--schema-path=\"{Path.GetFullPath(schemaDir)}\"");
+            }
+
+            // Add package schema directories
+            codegenArgs.AddRange(FindDirInPackages(SchemaPackageDir)
+                .Select(directory => $"--schema-path=\"{directory}\""));
+
+            // Schema Descriptor
+            codegenArgs.Add($"--descriptor-dir=\"{toolsConfig.DescriptorOutputDir}\"");
```

Comment:
This could be in the initial list right?

Reply:
err - yes it could